src/app/modules/main/stickers/module.nim

```diff
@@ -163,6 +160,21 @@ method estimate*(self: Module, packId: string, address: string, price: string, u
 method addRecentStickerToList*(self: Module, sticker: StickerDto) =
   self.view.addRecentStickerToList(initItem(sticker.hash, sticker.packId, sticker.url))
 
+method getRecentStickers*(self: Module) =
+  let data = self.controller.getRecentStickers()
```

Comment:
I am a bit confused about this part. There are 2 service functions: getRecentStickers() and loadRecentStickers(). I see that getRecentStickers() calls status_stickers.recent() in a sync way.  loadRecentStickers() calls status_stickers.recent() in an async way. So I am wondering why you call here the same function synchronously first and then asynchronously.

Reply:
Two things:

1. I'll add a comment to clarify this, but basically we first check if we already have recent stickers. `getRecentStickers` just returns the already loaded ones. If we have already loaded them, we're good. If not, **then** we load them asynchronously.
2. I noticed there's also a bug, we return to early here in L167. So in practice this means, the first itme this function is called, we load them asynchronously, and from that point on we get them from cache

Reply:
Actually, I stand corrected. Sticker service doesn't actually cache these. Gif service does.
Will fix that.

Thanks for pointing this out @endulab !

Reply:
Alright, on another note:

> I'll add a comment to clarify this, but basically we first check if we already have recent stickers. getRecentStickers just returns the already loaded ones. If we have already loaded them, we're good. If not, then we load them asynchronously.

For this particular data  (also installed sticker packs, it makes sense to always simply get it asynchronously... Otherwise the list will never update after initial load (if it wasn't empty then)

Will change this.

Reply:
So in other words:

> Actually, I stand corrected. Sticker service doesn't actually cache these. Gif service does.
Will fix that.

This was by design lol